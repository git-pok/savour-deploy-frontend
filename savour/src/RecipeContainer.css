.RecipeContainer-recipe {
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 300px;
    height: 100%;
    padding: 10px 10px;
    margin: 40px 20px;
    background-color: rgb(78, 42, 42);
    border-radius: 10px;
}

.RecipeContainer-recipe img {
    width: 100%;
    margin: 5px auto 5px auto;
    border-radius: 10px;
}
  
.RecipeContainer-recipe-p {
    display: block;
    margin: 5px auto 5px auto;
    text-align: left;
}
  
.RecipeContainer-recipe h3 {
    display: block;
    margin: 5px auto 5px auto;
    text-align: center;
    color: rgb(255, 255, 255);
}

.RecipeContainer-recipe a:link {
    display: block;
    margin: 5px auto 5px auto;
    text-align: left;
    text-decoration: none;
    /* color: rgb(251, 4, 4); */
    color: rgb(255, 255, 255);
}

.RecipeContainer-recipe a:hover h3 {
    /* color: #96a0a7; */
    color: rgb(223, 216, 85);
}