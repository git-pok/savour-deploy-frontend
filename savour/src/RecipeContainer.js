import { useContext } from 'react';
import { Link } from 'react-router-dom';
import UserContext from './context/UserContext.js';
import ShowHideBtnAndText from './ShowHideBtnAndText.js';
import './RecipeContainer.css';

/**
 * RecipeContainer
 * Props: showHide, recipeArray
 * Renders div with recipe content.
*/
const RecipeContainer = ({ showHide=false, recipeArray=[] }) => {
  const { usrData, setUsrData } = useContext(UserContext);

  return (
    <>
      {
        recipeArray.length ?
        recipeArray.map(recipe => (
          <div style={
              recipe.fav_user_ids &&
              recipe.fav_user_ids.includes(usrData.userId)
              ? { border: "2px solid green" }
              : null
            }
            key={recipe.id}
            className="RecipeContainer-recipe">
            <Link exact="true" to={`/recipes/${recipe.id}`}>
              <h3>{recipe.name}</h3>
            </Link>
            <h3>By {recipe.author}</h3>
            <img src={recipe.image}></img>
            { showHide ?
              <ShowHideBtnAndText paragraphArr={[
                {text: `Prep Time: ${recipe.prep_time}`},
                {text: `Cook Time: ${recipe.cook_time}`},
                {text: `Rating: ${recipe.rating}`},
                {text: `Difficulty: ${recipe.level}`},
                {text: `${recipe.description}`}
              ]} /> :
              <>
                <p className="RecipeContainer-recipe-p">Prep Time: {recipe.prep_time}</p>
                <p className="RecipeContainer-recipe-p">Cook Time: {recipe.cook_time}</p>
                <p className="RecipeContainer-recipe-p">{recipe.description}</p>
              </>
            }
          </div>
        ))
        : <h1>NO RECIPES</h1>
      }
    </>
  );
}

export default RecipeContainer;