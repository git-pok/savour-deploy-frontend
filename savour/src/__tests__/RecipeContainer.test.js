import RecipeContainer from '../RecipeContainer.js';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import UserContext from '../context/UserContext.js';


const usrData = {
  userUsername: "fvin",
  userId: 12
};


test("renders RecipeContainer without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{usrData}}>
        <RecipeContainer />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

test("renders snapshot without crashing", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{usrData}}>
        <RecipeContainer />
      </UserContext.Provider>
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});