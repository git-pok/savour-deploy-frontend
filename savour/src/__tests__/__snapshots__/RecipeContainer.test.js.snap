// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot without crashing 1`] = `
<DocumentFragment>
  <h1>
    NO RECIPES
  </h1>
</DocumentFragment>
`;
